src/pointer/set/CarManager.java

```diff
@@ -0,0 +1,213 @@
+                    Boolean.parseBoolean(params[1]));
+            return;
+        }
+
+        color = Color.fromString(params[2]);
+        BodyType bodyType = BodyType.fromString(params[3]);
+
+        if (color != null && bodyType != null) {
+            changer.changeCarsWithSmallDiameter(Float.parseFloat(param), params);
+        }
+
+        try {
+            changer.changeWheels(Float.parseFloat(param), Float.parseFloat(params[1]),
+                    TireType.fromString(params[2]));
+        } catch (NumberFormatException e) {
+            throw e;
+        } catch (IllegalArgumentException e) {
+            throw new IllegalArgumentException("Tire type is wong. Use: " +
+                    Arrays.toString(TireType.values()), e);
+        }
+
+        System.out.println("Parameter '" + param + "' isn't allowed. Allowed 'with'/'without', one of colors: "
+                + Arrays.toString(Color.values()));
+    }
+
+    void remove(String[] params) throws IllegalArgumentException {
+
+        switch (params.length) {
+            case 1:
+                Color color = Color.fromString(params[0]);
+
+                if (color == null) {
+                    throw new IllegalArgumentException("Parameter '" + params[0] +
+                            "' isn't allowed. Allowed colors: " + Arrays.toString(Color.values()));
+                }
+
+                remover.remove(color);
+                break;
+
+            case 2:
+                remover.remove(Float.parseFloat(params[0]), Float.parseFloat(params[1]));
+                break;
+
+            case 3:
+                BodyType bodyType = BodyType.fromString(params[0]);
+
+                if (bodyType == null) {
+                    throw new IllegalArgumentException("Parameter '" + params[0] +
+                            "' isn't allowed. Allowed body types: " + Arrays.toString(BodyType.values()));
+                }
+
+                remover.remove(bodyType, Float.parseFloat(params[1]), Float.parseFloat(params[2]));
+                break;
+        }
+    }
+
+    private void searchByOneParameter(String parameter) throws NumberFormatException {
+        if (parameter == null || "".equals(parameter)) {
+            System.out.println("Wrong parameters count. Expected one parameter.");
+            return;
+        }
+
+        if ("indexes".equals(parameter)) {
+            System.out.println(search.getIndexes());
+            return;
+        }
+
+        Color color = Color.fromString(parameter);
+
+        if (color != null) {
+            search.find(color);
+            return;
+        }
+
+        BodyType bodyType = BodyType.fromString(parameter);
+
+        if (bodyType != null) {
+            search.find(bodyType);
+            return;
+        }
+
+        search.find(Float.parseFloat(parameter));
+    }
+
+    private void searchByTwoParameters(String color, String diameter) throws IllegalArgumentException {
```

Comment:
what is this name? I can only see from parameters what it does

Reply:
it is private